scrypto/src/core/invocation.rs

```diff
@@ -1,102 +1,151 @@
 use sbor::rust::string::*;
 use sbor::rust::vec::Vec;
 use sbor::*;
+use strum::*;
 
+use crate::component::ComponentAddress;
 use crate::component::PackageAddress;
-use crate::engine::types::RENodeId;
+use crate::engine::types::{ComponentId, RENodeId};
 
-#[derive(Debug, Clone, Eq, PartialEq, TypeId, Encode, Decode)]
-pub enum FnIdent {
-    Function(FunctionIdent),
-    Method(ReceiverMethodIdent),
-}
-
-impl Receiver {
-    pub fn node_id(&self) -> RENodeId {
-        match self {
-            Receiver::Consumed(node_id) | Receiver::Ref(node_id) => *node_id,
-        }
-    }
+#[derive(Debug, Clone, Eq, PartialEq, Hash, TypeId, Encode, Decode)]
+pub struct ScryptoFunctionIdent {
+    pub package_address: PackageAddress,
+    pub blueprint_name: String,
+    pub function_name: String,
 }
 
 #[derive(Debug, Clone, Eq, PartialEq, Hash, TypeId, Encode, Decode)]
-pub enum FunctionIdent {
-    Scrypto {
-        package_address: PackageAddress,
-        blueprint_name: String,
-        ident: String,
-    },
-    Native(NativeFunction),
+pub struct NativeFunctionIdent {
+    pub blueprint_name: String,
+    pub function_name: String,
```

Comment:
What's the reasoning to keep these as strings? Seems like we lose the nice type checking and possibly integrating with args in the future.

Reply:
Two reasons:
- Decouple the interface from engine implementation
- Be consistent with Scrypto invocation

We're not losing compiler check since we immediately convert it into an enum upon `invoke_native` sys calls. 

The only downside of using string as discriminator is performance loss. We will need to address this separately if it proves to be a problem.